src/spatialdata/_core/spatialdata.py

```diff
@@ -1216,25 +829,49 @@ def read(file_path: str, selection: tuple[str] | None = None) -> SpatialData:
         return read_zarr(file_path, selection=selection)
 
     @property
-    def images(self) -> dict[str, SpatialImage | MultiscaleSpatialImage]:
+    def images(self) -> Images:
         """Return images as a Dict of name to image data."""
         return self._images
 
+    @images.setter
```

Comment:
This is equivalent to a `dict.extend()`, while if I called a `sdata.images = new_images` I would expect a complete replacemnet of the dict. I would change this to reflect this expected behavior. Same for labels, points and shapes.